py3status/py3.py

```diff
@@ -1289,15 +1291,37 @@ def request(
         if timeout is None:
             timeout = getattr(self._py3status_module, "request_timeout", 10)
 
+        if retry_times is None:
+            retry_times = getattr(self._py3status_module, "request_retry_times", 3)
+
+        if retry_wait is None:
+            retry_wait = getattr(self._py3status_module, "request_retry_wait", 2)
+
         if "User-Agent" not in headers:
             headers["User-Agent"] = "py3status/{} {}".format(version, self._uid)
 
-        return HttpResponse(
-            url,
-            params=params,
-            data=data,
-            headers=headers,
-            timeout=timeout,
-            auth=auth,
-            cookiejar=cookiejar,
-        )
+        def get_http_response(url, params, data, headers, timeout, auth, cookiejar):
+            return HttpResponse(
+                url,
+                params=params,
+                data=data,
+                headers=headers,
+                timeout=timeout,
+                auth=auth,
+                cookiejar=cookiejar,
+            )
+
+        for n in range(1, retry_times):
+            try:
+                return get_http_response(
+                    url, params, data, headers, timeout, auth, cookiejar
```

Comment:
(Simplify) Use `return get_http_response()` instead of passing in same parameters twice. (?)

Reply:
indeed